buildstockbatch/postprocessing.py

```diff
@@ -348,14 +434,25 @@ def combine_results(fs, results_dir, cfg, do_timeseries=True):
             all_ts_cols.difference_update(all_ts_cols_sorted)
             all_ts_cols_sorted.extend(sorted(all_ts_cols))
             logger.info(f"Got {len(all_ts_cols_sorted)} columns")
+            logger.info(f"The columns are: {all_ts_cols_sorted}")
         else:
-            logger.info("There are no timeseries files.")
+            logger.warning("There are no timeseries files for any upgrades.")
             do_timeseries = False
 
     results_df_groups = results_df.groupby('upgrade')
-    upgrade_start = 1 if cfg['baseline'].get('skip_sims', False) else 0
-    upgrade_end = len(cfg.get('upgrades', [])) + 1
-    for upgrade_id in range(upgrade_start, upgrade_end):
+    upgrade_list = get_upgrade_list(cfg)
+    partition_columns = cfg.get('postprocessing', {}).get('partition_columns', [])
+    partition_columns = [c.lower() for c in partition_columns]
+    df_partition_columns = [f'build_existing_model.{c}' for c in partition_columns]
+    missing_cols = set(df_partition_columns) - set(all_schema_dict.keys())
+
+    if missing_cols:
+        df_partition_columns = list(set(df_partition_columns) - set(missing_cols))
+        logger.info(f"The following patition columns are not available on the results.json: {missing_cols}")
+        logger.info(f"Only using these as the partition column:s {df_partition_columns}")
```

Comment:
Also seems like this should error out. This should probably be a pre run validation if possible. And if we do keep this, aren't the order of the partition columns important? (i.e. state then county) This is using `set` which can return the columns in any order.

Reply:
I have included validation for this, so, if the input partition_columns specified in the yaml is wrong, it will fail validation. The check here is for cases when the columns are missing somehow in the results_json (for example all sims failed or some bugs in the simulation / resstock). In that case, I think you are right, we just raise an error and crash instead of silently continuing on.